os/arch/arm/src/amebad/amebad_serial.c

```diff
@@ -374,6 +374,25 @@ static u32 uart_index_get(PinName tx)
 	return 3;
 }
 
+static void LOGUART_PutChar_RAM(u8 c)
+{
+	UART_TypeDef* LOGUART = UART2_DEV;
+	u32 CounterIndex = 0;
+
+	if (ConfigDebugClose == 1)
+		return;
+
+	while(1) {
+		CounterIndex++;
+		if (CounterIndex >=6540)
+			break;
+
+		if (LOGUART->LSR & RUART_LINE_STATUS_REG_THRE)
```

Comment:
Ok, here it checks status of holding register, we don't have to disable interrupt first, and enable it when holding register is empty to prevent data corruption by data in uart buffer?